Fix for "GetItNow" action behaviour

swmitra opened the pull request:
When user clicks on "GetItNow" it should perform one of these actions 

- Either initialises Auto-Update process after all the conditions verified. 
- Or if Auto-Update can't be initialised, then take the user to brackets.io.

However, we shouldn't perform both the actions. 

This PR fixes the faulty behaviour and ensures, we don't open brackets.io if we are able to start auto-update successfully. 

@shubhsnov @narayani28 Please review.

swmitra committed:
Fix for "GetItNow" action behaviour

Fix for "GetItNow" action behaviour

shubhsnov commented on src/extensions/default/AutoUpdate/main.js:
```diff
@@ -483,6 +483,9 @@ define(function (require, exports, module) {
             initiateAutoUpdate(updateParams);
+	    
+	    //Send a truthy value to ensure caller is informed about successful initialization of auto-update
+	    return true;
```
This seems okay to me. 

Just one nit, in UpdateNotification.js the _updateProcessHandler is actually an async function(internally calls initiateAutoUpdate), but  _defaultUpdateProcessHandler would work synchronously. The way the code is written seems to imply that the update was successfully handled, which is not the case even if the initialization was successful. Should we change the variable name to `initSuccess` in UpdateNotification.js?
On the otherhand, we can return a promise from here and resolve it with true or false accordingly to indicate actual success, but that would make things a bit complicated.

swmitra replied:
Understood. Will change the variable name in that case. It can't be async, or rather we can't wait till success of auto-update process itself, because that would break the continuity of user journey in the workflow. No point in launching brackets.io after 5 minutes if download has failed.

shubhsnov replied:
Oh yes, definitely. What I meant was to handle the initialization failure scenario separately from update success/failure scenario, if required but through a promise since that will be more reliable. As for update failure, we're already telling the user to go to the site if the update fails [here](https://github.com/adobe/brackets/blob/master/src/extensions/default/AutoUpdate/main.js#L247).

swmitra replied:
Changed `success` to `initSuccess` now.

swmitra committed:
Change variable success to initSuccess 

Make the variable name more explicit to depict it's to track update process initialisation success and not the success status of update process itself.
